src/component/Text.tsx

```diff
@@ -160,132 +151,106 @@ const getWordsWithoutCalculate = (children: React.ReactNode): Array<Words> => {
   return [{ words }];
 };
 
-const getWordsByLines = (props: Props, needCalculate: boolean) => {
+const getWordsByLines = (props: Props) => {
   // Only perform calculations if using features that require them (multiline, scaleToFit)
   if ((props.width || props.scaleToFit) && !Global.isSsr) {
     let wordsWithComputedWidth: Array<WordWithComputedWidth>, spaceWidth: number;
 
-    if (needCalculate) {
```

Comment:
does the `useMemo` account for `needCalculate`? Looks like it does, just confirming

Reply:
I think it does, previously the diffing was done by keeping prev props in state and in the subsequent re-renders, the code was comparing new props to older ones to see the need of recalculating, but now that whole logic is done by React itself.

I also refactored the utility function types so each function takes only desired props, which makes it easy to narrow down the props that is necessary to trigger the recalculation.